### PR TITLE
GODRIVER-2442 replace macos-1014 with macos-1015 in Evergreen

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary
- Replace macos-1014 with macos-1015 in Evergreen
- Remove unnecessary build of libmongocrypt in run-tests.sh.
- Get libmongocrypt from published binaries.

# Background & Motivation
This is motivated by FLE 2 test failures on macOS 10.14. FLE 2 crypto primitives require macOS 10.15 in libmongocrypt: [MONGOCRYPT-385](https://jira.mongodb.org/browse/MONGOCRYPT-385).

macOS 10.14 is no longer supported by Apple. I could not find a source from Apple, but I see it listed on [Wikipedia](https://en.wikipedia.org/wiki/MacOS_Mojave) under Support Status in the sidebar. MongoDB server 6.0 [drops support of 10.14](https://docs.google.com/document/d/144UY1N3rclma-0XXbfYIfd627m88wJqJZ84nEVm1Sko/edit#heading=h.nspqzc7ixrey).

libmongocrypt does not currently compile on macOS 10.15 hosts (see MONGOCRYPT-443). Downloading a binary from the macOS 11.00 host is a proposed workaround until MONGOCRYPT-443 is resolved.

Alternatively, I could skip the FLE 2 tests on macOS.

Failing Windows tasks are expected to be resolved by https://github.com/mongodb/mongo-go-driver/pull/981.

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -101,6 +101,25 @@ functions:
              cp ./install/libmongocrypt/bin/mongocrypt.dll c:/libmongocrypt/bin
              cp ./install/libmongocrypt/include/mongocrypt/*.h c:/libmongocrypt/include
              export PATH=$PATH:/cygdrive/c/libmongocrypt/bin
+          elif [ "Darwin" = "$(uname -s)" ]; then
+             # TODO (GODRIVER-2442): Once libmongocrypt compiles on macOS 10.15, remove this elif block.
```
Comment (benjirewis) [inline .evergreen/config.yml:105]: So we won't close GODRIVER-2442 until we can remove this `elif` block?

File: .evergreen/config.yml [hunk 2]
```diff
@@ -101,6 +101,25 @@ functions:
...
+             # The following URL is published from the upload-all task in the libmongocrypt Evergreen project.
+             curl https://mciuploads.s3.amazonaws.com/libmongocrypt/all/master/$LIBMONGOCRYPT_COMMIT/libmongocrypt-all.tar.gz -o libmongocrypt-all.tar.gz
+             tar -xf libmongocrypt-all.tar.gz
+             cd ..
+             mv libmongocrypt-all/macos/include ./install/libmongocrypt
+             mv libmongocrypt-all/macos/lib ./install/libmongocrypt
+             rm -rf libmongocrypt-all
+             rm libmongocrypt-all.tar.gz
+             # Fix prefix in pkg-config prefix path.
+             sed -i "" -E "s+prefix=.*+prefix=$(pwd)/install/libmongocrypt+" ./install/libmongocrypt/lib/pkgconfig/libmongocrypt.pc
```
Comment (benjirewis) [inline .evergreen/config.yml:121]: Love `sed` haha. What exactly is this "fixing" the prefix path to?

**Review by benjirewis — COMMENTED** [review]
Switching the MacOS version we're running against seems better than skipping the FLE 2 tests when running against MacOS IMO (especially since MacOS 10.14 support seems to have been dropped by Apple themselves). There are still some sharded cluster failures on the MacOS 10.15 tests attached to this PR; are they all timeouts unrelated to CSFLE? There seem to be more than are currently in the waterfall. It may actually be helpful to make those flaky test timeouts less flaky so we can continue the investigation in GODRIVER-2266.

File: .evergreen/config.yml (hunk 2 above)
Comment (kevinAlbs) [reply .evergreen/config.yml:121]: The package config prefix includes the install path from the host that builds.

This is the contents of this tarball's macos/lib/pkgconfig/libmongocrypt.pc:

```
Name: mongocrypt
Description: The libmongocrypt client-side field level encryption library.
Version: 1.5.0-rc0
Requires: 
Requires.private: 
prefix=/System/Volumes/Data/data/mci/253c2ab5479da6c90e38aaa963434d83/install/libmongocrypt
includedir=${prefix}/include/mongocrypt
libdir=${prefix}/lib
Libs: -L${libdir} -lmongocrypt
Cflags: -I${includedir}
```

The path is included in the output of pkg-config:
```
% export PKG_CONFIG_PATH=./macos/lib/pkgconfig/ 
% pkg-config --cflags --libs libmongocrypt
-I/System/Volumes/Data/data/mci/253c2ab5479da6c90e38aaa963434d83/install/libmongocrypt/include/mongocrypt -L/System/Volumes/Data/data/mci/253c2ab5479da6c90e38aaa963434d83/install/libmongocrypt/lib -lmongocrypt
```

**Review by kevinAlbs — COMMENTED** [review]
> There are still some sharded cluster failures on the MacOS 10.15 tests attached to this PR; are they all timeouts unrelated to CSFLE?

AFAICT they are unrelated to CSFLE. The FLE 2 tests do not run on server versions < 6.0 and 
many of those task failures are on older server versions.

File: .evergreen/config.yml (hunk 2 above)
Comment (benjirewis) [reply .evergreen/config.yml:121]: Cool thanks for the explanation.

**Review by benjirewis — APPROVED** [review]

**Review by matthewdale — APPROVED** [review]
Looks good 👍

**Pull request merged**